solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -328,25 +326,26 @@ protected NamedList loadConfigSetFlags(CoreDescriptor cd, SolrResourceLoader loa
   public abstract void setConfigMetadata(String configName, Map<String, Object> data) throws IOException;
 
   /**
-   * Get the config metadata
+   * Get the config metadata (mutable) or null
```

Comment:
Instead of null, let's never return null?  This would be friendlier on the caller.  It would only be not okay if somehow it was meaningful to have blank metadata, which seems weird to me.